enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,583 @@
+expected without any known issues.
+
+Even on Assisted Installer installed single control-plane node clusters it's
+trivial to add more workers by leveraging the Assisted Installer's day-2 worker
+installation capabilities (after some minor DNS configuration issues which will
+be improved by the Assisted Installer team, separately from this enhancement).
+
+OCP installations have an `infrastructures.config.openshift.io` CR
+automatically deployed by the installer. This CR has two topology fields in
+its status called "Control Plane Topology" and "Infrastructure Topology" which,
+as of today, may have one of three values:
+
+- `SingleReplica`
+- `HighlyAvailable`
+- `External` (this value is unrelated to this enhancement and is only mentioned here for completeness' sake)
+
+(See "see-also" enhancements links for more information about the topology
+fields and their possible values)
+
+The "Control Plane Topology" field is used by control-plane operators (such
+as the Cluster etcd Operator) to determine how many replicas they should give
+their various Deployments / StatefulSets. The value of this field in single
+control-plane node clusters is always simply `SingleReplica`.
+
+The "Infrastructure Topology" field is used by infrastructure operators
+(such as the Cluster Ingress Operator or Cluster Monitoring Operator) to
+determine how many replicas they should give their various Deployments /
+StatefulSets. The value of this field is a function of how many workers
+were present during installation.
+
+On "none"-platform single control-plane node clusters, when adding
+workers, the resulting cluster has an issue with the behavior of the ingress
+pod, the following paragraphs explain the background for this issue and the
+issue itself.
+
+One of the benefits of installing single-node clusters is the simplicity of not
+having to deal with load-balancing and virtual IPs, as these don't provide much
+value when there's only a single node behind them.
+
+As a result, current common ways of installing single control-plane node
+clusters today (mainly the Assisted Installer) avoid the usage of load
+balancers or virtual IPs for API and ingress. There has also been some recent
+effort to determine how single control-plane node cluster deployments on clouds
+may be adjusted in order to reduce their costs, and one of the conclusions is
+that getting rid of the load balancer installed by default by the IPI installer
+results in major cost savings.
+
+A user installing a single control-plane node cluster on "none"-platform will
+be tempted to simply point their DNS entries directly at the IP address of the
+single node that they just installed.
+
+Similarly, in the Assisted Installer, the user is able to complete the
+installation without needing to define any DNS entries. This is currently made
+possible by injecting a `MachineConfig` manifest targeting the "master" pool
+containing configuration for a dnsmasq server. The dnsmasq server is configured
```

Comment:
I wasn't aware of that; seems like a nontrivial bit.  Where is that code?

Reply:
In assisted installer. We just inject a manifest via standard installer apis

Reply:
@cgwalters My impression is that this is really background info, and doesn't affect the change proposed here. Is that right?

Reply:
Not sure whether that question is directed towards me or not, but yeah, I was just providing it as background information

Reply:
The question was for @cgwalters. I don't think there's any impact, but want to make sure before we merge this.

Reply:
No, not a blocking concern.

Reply:
Thanks, Colin